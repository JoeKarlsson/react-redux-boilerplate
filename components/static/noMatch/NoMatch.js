import React from 'react';
import styles from './NoMatch.scss';

class NoMatch extends React.Component {
  render() {
    return (
      <div className={styles.noMatch}>
        No Page Found
      </div>
    );
  }
}

export default NoMatch;
