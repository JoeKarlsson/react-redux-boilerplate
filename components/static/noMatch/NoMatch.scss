$pink: pink;

.noMatch {
  background-color: $pink;
}
