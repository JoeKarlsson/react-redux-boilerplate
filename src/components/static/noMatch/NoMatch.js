import React from 'react';
import styles from './NoMatch.scss';

const NoMatch = () => {
	return (
		<div className={styles.noMatch}>
      No Page Found
		</div>
	);
};

export default NoMatch;
