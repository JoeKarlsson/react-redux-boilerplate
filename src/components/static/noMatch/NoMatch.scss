$pink: pink;

.NoMatch {
	background-color: $pink;
}
