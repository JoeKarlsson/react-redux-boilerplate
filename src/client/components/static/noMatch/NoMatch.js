import React from 'react';
import './NoMatch.scss';

const NoMatch = () => {
	return (
		<div className="NoMatch">
      No Page Found
		</div>
	);
};

export default NoMatch;
