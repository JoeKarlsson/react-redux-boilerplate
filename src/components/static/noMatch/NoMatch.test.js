import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import NoMatch from './NoMatch';

describe('NoMatch Page', () => {
	let wrapper;
	let inst;

	beforeEach(() => {
		wrapper = shallow(
			<NoMatch />,
		);
		inst = wrapper.instance();
	});

	describe('rendering', () => {
		describe('initial state', () => {
			it('match the snapshot', () => {
				const component = renderer.create(
					<NoMatch />,
				);
				const tree = component.toJSON();
				expect(tree).toMatchSnapshot();
			});
			it('is rendered correctly', () => {
				expect(wrapper).toHaveLength(1);
			});
			it('should render without throwing an error', () => {
				expect(wrapper.text()).toEqual('No Page Found');
			});
			it('should be selectable by the class `ChatDrawer`', () => {
				expect(wrapper.is('.NoMatch')).toBe(true);
			});
			it('should mount in the full DOM', () => {
				expect(wrapper.find('.NoMatch').length).toBe(1);
			});
			it('should have correct inital state', () => {
				const initialState = inst.state;
				const expectedIntialState = null;
				expect(initialState).toBe(expectedIntialState);
			});
			it('should not have any inital props', () => {
				const initialProps = inst.props;
				const expectedProps = {};
				expect(initialProps).toEqual(expectedProps);
			});
		});
	});
});
