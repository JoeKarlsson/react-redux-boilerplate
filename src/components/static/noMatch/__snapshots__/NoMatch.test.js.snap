// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch page snapshot matches 1`] = `
<div
  className="noMatch"
>
  No Page Found
</div>
`;
